app/actions/websocket.ts

```diff
@@ -641,17 +640,28 @@ function handleUserRemovedEvent(msg: WebSocketMessage) {
             const currentChannelId = getCurrentChannelId(state);
             const currentTeamId = getCurrentTeamId(state);
             const currentUser = getCurrentUser(state);
-            const actions: Array<GenericAction> = [{
-                type: ChannelTypes.CHANNEL_MEMBER_REMOVED,
-                data: {
-                    channel_id: msg.broadcast.channel_id,
-                    user_id: msg.data.user_id,
-                },
-            }];
+            const actions: Array<GenericAction> = [];
+            if (msg.data.user_id) {
+                actions.push({
+                    type: ChannelTypes.CHANNEL_MEMBER_REMOVED,
+                    data: {
+                        channel_id: msg.broadcast.channel_id,
+                        user_id: msg.data.user_id,
+                    },
+                });
+            } else if (msg.broadcast.user_id) {
+                actions.push({
+                    type: ChannelTypes.CHANNEL_MEMBER_REMOVED,
+                    data: {
+                        channel_id: msg.data.channel_id,
+                        user_id: msg.broadcast.user_id,
+                    },
+                });
+            }
 
             const channel = channels[currentChannelId];
 
-            if (msg.data.user_id !== currentUser.id) {
+            if (msg.data.user_id && msg.data.user_id !== currentUser.id) {
```

Comment:
why do we need to check for the existence of `msg.data.user_id` in any case this can be
```suggestion
            if (msg.data?.user_id !== currentUser.id) {
```

Reply:
It handle 2 kind of messages, the message of "user was removed to from the channel" for the people in the channel, and one for a broadcast message sent to the person who is no longer a member of the channel. So it can be `msg.data.user_id` or `msg.broadcast.user_id`

Reply:
changed